mempool/src/core_mempool/transaction_store.rs

```diff
@@ -237,32 +255,55 @@ impl TransactionStore {
     }
 
     /// Maintains the following invariants:
-    /// - All transactions of a given account that are sequential to the current sequence number
+    /// - All transactions of a given non-CRSN account that are sequential to the current sequence number
     ///   should be included in both the PriorityIndex (ordering for Consensus) and
     ///   TimelineIndex (txns for SharedMempool).
+    /// - All transactions of a given CRSN account that are greater than the account's min_nonce
+    ///   should be included in both the PriorityIndex and TimelineIndex.
     /// - Other txns are considered to be "non-ready" and should be added to ParkingLotIndex.
     fn process_ready_transactions(
         &mut self,
         address: &AccountAddress,
-        current_sequence_number: u64,
+        crsn_or_seqno: AccountSequenceNumber,
     ) {
         if let Some(txns) = self.transactions.get_mut(&address) {
-            let mut sequence_number = current_sequence_number;
-            while let Some(txn) = txns.get_mut(&sequence_number) {
-                self.priority_index.insert(txn);
-
-                if txn.timeline_state == TimelineState::NotReady {
-                    self.timeline_index.insert(txn);
+            let mut min_seq = crsn_or_seqno.min_seq();
+
+            match crsn_or_seqno {
+                AccountSequenceNumber::CRSN { min_nonce, size } => {
+                    for i in min_nonce..size {
+                        if let Some(txn) = txns.get_mut(&i) {
+                            self.priority_index.insert(txn);
+
+                            if txn.timeline_state == TimelineState::NotReady {
+                                self.timeline_index.insert(txn);
+                            }
+
+                            // Remove txn from parking lot after it has been promoted to
+                            // priority_index / timeline_index, i.e., txn status is ready.
+                            self.parking_lot_index.remove(txn);
+                            min_seq = i;
+                        }
+                    }
```

Comment:
Is it possible that `txns.len < size`?  Should you just break at that point on the loop?

Reply:
It could be possible that `txns.len < size` however, since we're getting these transactions by their sequence nonce, we need to check for any transactions that could be processed in the window (and they don't need to be contiguous) so we need to look for each possibly-processable transaction in the current sequence nonce window.